### PR TITLE
Add `MySQL` backup/restore process by using KubeStash 

### PR DESCRIPTION
**Pull request by anisurrahman75**
Add auto backup

### CODE REVIEW

File: docs/guides/mysql/backup/kubestash/auto-backup/index.md [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+Let's create the `MySQL` we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/kubedb/docs/raw/{{< param "info.version" >}}/docs/guides/mysql/backup/kubestash/auto-backup/examples/sample-mysql.yaml
+mysql.kubedb.com/sample-mysql created
+```
+
+**Verify BackupConfiguration**
+
+If everything goes well, KubeStash should create a `BackupConfiguration` for our MySQL in demo namespace and the phase of that `BackupConfiguration` should be `Ready`. Verify the `BackupConfiguration` object by the following command,
```
Comment (AshrafulHaqueToni) [inline docs/guides/mysql/backup/kubestash/auto-backup/index.md:242]: If everything is set up correctly, KubeStash will create a `BackupConfiguration` for our MySQL instance in the `demo` namespace. The phase of this `BackupConfiguration` should be `Ready`. You can verify the `BackupConfiguration` object by running the following command,

File: docs/guides/mysql/backup/kubestash/auto-backup/index.md [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+```bash
+$ kubectl get backupconfiguration -n demo
+NAME                      PHASE   PAUSED   AGE
+appbinding-sample-mysql   Ready            2m50m
+```
+
+Now, let’s check the YAML of the `BackupConfiguration`.
+
+```bash
+$ kubectl get backupconfiguration -n demo appbinding-sample-mysql  -o yaml
```
Comment (AshrafulHaqueToni) [inline docs/guides/mysql/backup/kubestash/auto-backup/index.md:253]: add the resulted yaml of this command

**Review by AshrafulHaqueToni — CHANGES_REQUESTED** [review]

File: docs/guides/mysql/backup/kubestash/logical/index.md [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+At first, we need to create a secret with a Restic password for backup data encryption.
+
+**Create Secret:**
+
+Let's create a secret called `encrypt-secret` with the Restic password,
+
+```bash
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ kubectl create secret generic -n demo encrypt-secret \
+    --from-file=./RESTIC_PASSWORD \
```
Comment (tapojit047) [inline docs/guides/mysql/backup/kubestash/logical/index.md]: Remove backslash at the end of the command

File: docs/guides/mysql/backup/kubestash/logical/index.md [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+```bash
+$ kubectl get repository -n demo gcs-mysql-repo
+NAME                    INTEGRITY   SNAPSHOT-COUNT   SIZE    PHASE   LAST-SUCCESSFUL-BACKUP   AGE
+gcs-mysql-repo          true        1                806 B   Ready   8m27s                    9m18s
+```
+
+At this moment we have one `Snapshot`. Run the following command to check the respective `Snapshot` which represents the state of a backup run for an application.
+
+```bash
+$ kubectl get snapshots -n demo -l=kubestash.com/repo-name=gcs-demo-repo
```
Comment (tapojit047) [inline docs/guides/mysql/backup/kubestash/logical/index.md]: kubectl get snapshots -n demo -l=kubestash.com/repo-name=gcs-mysql-repo

File: docs/guides/mysql/backup/kubestash/customization/index.md [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: MySQL
+    namespace: demo
+    name: restored-mysql
+  dataSource:
+    repository: gcs-mysql-repo
+    snapshot: latest
```
Comment (tapojit047) [inline docs/guides/mysql/backup/kubestash/customization/index.md]: we should provide a specific value for snapshot, right?

`snapshot: gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725258600`

File: docs/guides/mysql/backup/kubestash/application-level/index.md [hunk 6]
```diff
@@ -0,0 +1,761 @@
...
+```bash
+$ kubectl get repository -n demo gcs-mysql-repo
+NAME                    INTEGRITY   SNAPSHOT-COUNT   SIZE    PHASE   LAST-SUCCESSFUL-BACKUP   AGE
+gcs-mysql-repo          true        1                806 B   Ready   8m27s                    9m18s
+```
+
+At this moment we have one `Snapshot`. Run the following command to check the respective `Snapshot` which represents the state of a backup run for an application.
+
+```bash
+$ kubectl get snapshots -n demo -l=kubestash.com/repo-name=gcs-demo-repo
```
Comment (tapojit047) [inline docs/guides/mysql/backup/kubestash/application-level/index.md]: `gcs-demo-repo` --> `gcs-mysql-repo`

**Pull request merged**